flink-connectors/flink-connector-kafka/src/main/java/org/apache/flink/streaming/connectors/kafka/FlinkKafkaProducer.java

```diff
@@ -647,8 +646,8 @@ private FlinkKafkaProducer(
                 new FlinkKafkaProducer.TransactionStateSerializer(),
                 new FlinkKafkaProducer.ContextStateSerializer());
 
+        checkArgument(!"".equals(defaultTopic),"defaultTopic is empty");
```

Comment:
maybe we can use
'
checkArgument(
                !org.apache.flink.util.StringUtils.isNullOrWhitespaceOnly(defaultTopic), "defaultTopic cannot be null or empty string");
'
to check